### DISCUSSION TITLE
Label (or tag?) for threads with scripts examples

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#541] [Ideas]
I find the Scripts Collection in the documentation very useful. Maybe, an easy way to keep this idea active is using a label for the threads in the Discussion and Issues where scripts are created. So, the user can see a "scripts collection" from the forum simply by filtering by the label. 

I don't know what would be the most convenient way to implement that. Maybe, a bot that inserts a label when an user comments with a hashtag: #scripts. 

What do you think?

**Comment by macmarrum** [comment]
This idea of tagging posts is already live for mindmap examples (aka show & tell) – https://github.com/freeplane/freeplane/discussions/categories/shared-mind-maps

It makes sense to take the same approach for scripts, i.e. a new Category could be created and users would simply select it on their posts.

**Comment by EdoFro** [comment]
I totally agree with the use of more labels to tag discussions. And I think it makes cense to use a bot to do this.

I've been reading about bots in github and have make some little tests but I'm not able to do complex things yet. For those you need ideally some javascript knowledge. If somebody has more experience with Github actions, maybe he/she can help.

**Comment by euu2021** [comment]
About labels X categories

I'm more into labels, because

- doing a quick search I found a ton of bots for labeling, but not much information about bots for categories. So, it's probably easier to automatize labels 
- this approach can be extended for other things, like making it part of the documentation workflow. Edo explained it on a previous thread, and I mentioned it on the old forum 
- labels are shared with issues (and some scripts are also posted in the issues section). Dimitry probably won't be happy with people messing with his categorization on the issues section.

**Reply by euu2021** [reply]
Complementing point 2 above: so... With labels this expansion would be easier, because there can multiple labels per issue, which is not the case for categories

**Reply by euu2021** [reply]
And, about the third point: I just noticed that there are no categories in the issues section, it's only labels. So it seems that labels are the only option, because that's the only way to capture scripts posted in the issues comments.

**Comment by euu2021** [comment]
So, a simple solution that doesn't need coding is simply using a keyword that can be used in the search. 

For example, I commented on this (https://github.com/freeplane/freeplane/discussions/415#discussioncomment-3339552) script by Edo, adding a short description after the keyword scritCollection. In the search I only this result (so, no false positives), with the bonus of having the description of the script in the search snippet:
![IMG_20220806_102731](https://user-images.githubusercontent.com/77707706/183251152-98859bf3-ac5c-4b9d-9931-4ac9a75a8ce2.jpg)

I like this. What do you think?

**Comment by dpolivaev** [comment]
I can add labels #script, #template and #mindmap.
I find labels consisting of more than one word unnecessarily complex until all possible one-word labels have been used.
What do you think?

**Reply by euu2021** [reply]
Right now, I'm preferring the solution of using a keyword to be searched later, but not relying on labels (https://github.com/freeplane/freeplane/discussions/541#discussioncomment-3339567). Because:

1. Possibility to include a description to appear in the snippet 
2. Possibility to use multiple times per thread
3. No need to create bots. This would be necessary in order to allow any user to make the labelling. 

Then, the documentation can link directly to the search query: https://github.com/freeplane/freeplane/search?q=scriptsCollection&type=discussions

**Reply by dpolivaev** [reply]
Unfortunately, it makes sense. I hate workarounds so much because they do not feel right, but there is no better choice.
How can we indicate that some word is used as a tag?

**Reply by euu2021** [reply]
---
I've made some tests, and found:
About my point 2 above (Possibility to use multiple times per thread), there's a problem though. The github search results shows only 1 result, even if the word appears multiple times in the thread. This is relevant, because sometimes there are multiple scripts per thread. 
I don't know what could be a non anoying workaround. 
An example of annoying workaround is: make each instance of the tag, afeter the first one, be numbered (eg exampleTag2; exampleTag3), so the user needs to make multiple searches in order to see all the scripts of the scripts collection.

---
Another finding:
Placing the tag at the beggining of the text gives more characters in the snippet, so that could be suggested as a good practice. Example:

![image](https://user-images.githubusercontent.com/77707706/183264713-5ee38a30-2961-439e-906f-df69ec6c50f4.png)

---
> How can we indicate that some word is used as a tag?

Do you mean, how the tag should appear in the comment? Do you think users will get confused with a comment that starts with the tag, like this:
![image](https://user-images.githubusercontent.com/77707706/183264834-972ee945-5aa3-4d23-9edf-ba3045589477.png)

**Comment by euu2021** [comment]
I think the ideal solution here would be using this thing called ChatOps (a bot that allows executing commands in the repository directly from a comment). So, the user makes a comment using a specific notation like `[/scriptsCollection: short script description...]`, and the bot grabs the content inside the [ ] and appends it to the end of the Scripts Collection document, through a commit, and links back to the original comment.
But, this is beyond my skill level, so, what I can propose is the non-programmer solution of searching the tag in the Github search.

---
About the limitation of the search showing only one instance of the tag per thread, I think this will only affect a small part of the cases, so it doesn't seem to be a dealbreaker to me. To attenuate the problem, I suggest:

- Using the tag scriptsCollection2 after the first instance of the thread. The same tag would be used for the third, fourth and following instances, too (ie, there would be no scriptsCollection3 tag) .
- On the documentation (where will be located the link to the tag search) put an explicit warning that, on that second tag search, there could be more scripts inside the thread that will nor appear in the search results.

---
If someone has any considerations about this approach, please write down below. Otherwise, I will follow by writing a draft of suggested guidelines to be followed when applying the tag.

**Comment by euu2021** [comment]
I noticed that `scriptDescription` could be a better tag, because it blends better with the original content, and avoids confusion in the user. Look an example at the first post of #548.
